internal/hsm/interface.go

```diff
@@ -0,0 +1,44 @@
+// MIT License
+// 
+// (C) Copyright [2022] Hewlett Packard Enterprise Development LP
+// 
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following conditions:
+// 
+// The above copyright notice and this permission notice shall be included
+// in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+// THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+// OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+// ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+// OTHER DEALINGS IN THE SOFTWARE.
+
+package hsm
+
+import (
+	"github.com/Cray-HPE/hms-base"
+)
+
+type HSMv2 struct {
+	HSMGlobals HSM_GLOBALS
+}
+
+type HSMProvider interface {
+	Init(glb *HSM_GLOBALS) error
+	Ping() error
+	ReserveComponents(compList []ReservationData) ([]*ReservationData,error)
+	ReleaseComponents(compList []ReservationData) ([]*ReservationData,error)
+	CheckDeputyKeys(comp []ReservationData) error
+	FillComponentEndpointData(hd map[string]*HsmData) []error
```

Comment:
'[]error' is the incorrect return type for FillComponentEndpointData() in state_manager.go. It should be 'error'.

Reply:
Good catch.   Fixed.

Reply:
The reason why it was []error is because its an array of ComponentEndpoint data. Im not sure it should be changed to a singleton.

Reply:
The function itself in state_manager.go is a singleton. This was just being changed here to match that.

As far as singleton vs array goes, the function isn't acting individually on each ComponentEndpoint. I think a single error is sufficient to say whether or not it successfully got info back from HSM.